src/app/shared/editor/components/sql-editor/sql-editor.component.ts

```diff
@@ -0,0 +1,63 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    EventEmitter,
+    Input,
+    OnChanges,
+    Output,
+    SimpleChanges,
+} from "@angular/core";
+import * as monaco from "monaco-editor";
+
+import { getMonacoNamespace, MonacoService } from "../../services/monaco.service";
+import { SQL_EDITOR_OPTIONS } from "./config-editor.events";
+import { MaybeNull } from "src/app/common/app.types";
+
+@Component({
+    selector: "app-sql-editor",
+    templateUrl: "./sql-editor.component.html",
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SqlEditorComponent implements OnChanges {
+    @Input() protected readonly EDITOR_OPTIONS: monaco.editor.IStandaloneEditorConstructionOptions = SQL_EDITOR_OPTIONS;
```

Comment:
Similarly, why @Input, and why protected?

Reply:
Protected - just copied from someplace, I will remove it.

Reply:
removed